core/src/main/scala/org/genivi/sota/core/campaigns/CampaignLauncher.scala

```diff
@@ -35,8 +41,22 @@ object CampaignLauncher {
     Future.successful(devices.map(_ -> Set(pkg.id)).toMap)
   }
 
-  def launch (deviceRegistry: DeviceRegistry, updateService: UpdateService, id: Campaign.Id, lc: LaunchCampaign)
-             (implicit db: Database, system: ActorSystem, ec: ExecutionContext): Future[List[Uuid]] = {
+  //scalastyle:off parameter.number
+  def sendMsg(namespace: Namespace, devices: Set[Uuid], pkgId: PackageId, updateId: Uuid,
+                      messageBus: MessageBusPublisher)
+                     (implicit db: Database, system: ActorSystem, ec: ExecutionContext, mat: ActorMaterializer)
```

Comment:
I don't think you need the `ActorMaterializer` actually do you need the `ActorSystem`?